life_expectancy/main.py

```diff
@@ -1,37 +1,54 @@
 """
 Script to load, clean and save data
 """
-# import sys
-
-# sys.path.append('/nfs/backup/wb_mci_crp/fcalves/assignments')
 
 import argparse
 import pathlib
-from life_expectancy.clean_data import clean_data
-from life_expectancy.data_loading import load_data, save_data
-
-
+import os
+from typing import Union
+import pandas as pd
+from life_expectancy import clean
+from life_expectancy import load
+from life_expectancy.region import Region
 
 PARENT_PATH = pathlib.Path(__file__).parent
-FILE_PATH = PARENT_PATH / 'data'
-INPUT_FILE_NAME = 'eu_life_expectancy_raw.tsv'
-OUTPUT_FILE_NAME = 'pt_life_expectancy.csv'
-INPUT_FILE_PATH = FILE_PATH / INPUT_FILE_NAME
-OUTPUT_FILE_PATH = FILE_PATH / OUTPUT_FILE_NAME
-
-
-def main(country: str = 'PT') -> None:
+DATA_PATH = PARENT_PATH / 'data'
+TSV_PATH = DATA_PATH / 'eu_life_expectancy_raw.tsv'
+CSV_PATH = DATA_PATH / 'pt_life_expectancy.csv'
+ZIPPED_FILE = DATA_PATH / 'eurostat_life_expect.zip'
+JSON_PATH = DATA_PATH / 'eurostat_life_expect.csv'
+
+def save_data(dataframe: pd.DataFrame, file_path: Union[str, pathlib.Path]) -> None:
+    """Function to save data, given a file_name and a path"""
+    dataframe.to_csv(file_path, index = False)
+
+def main(file_path: Union[str, pathlib.Path],
+    output_file: Union[str, pathlib.Path],
+    country: Region) -> None:
     """
         Function that loads, cleans and saves data
     """
-    data_df = load_data(INPUT_FILE_PATH)
-    clean_df = clean_data(df_expectancy = data_df, country = country)
-    save_data(dataframe = clean_df, file_path = OUTPUT_FILE_PATH)
-    return clean_df
+    file_format = os.path.splitext(file_path)[1]
```

Comment:
I would just make a suggestion here (based on a comment Fernando gave me): If you use pathlib.Path instead of strings, you can replace this function with path.suffix.

An example:
from pathlib import Path

file_path = Path("/path/to/file.txt")
file_extension = file_path.suffix

print(file_extension)  # Output: ".txt"